Training24/Test.c

```diff
@@ -1,14 +1,105 @@
 // ------------------------------------------------------------------------------------------------
 // Training ~ A training program for new joiners at Metamation, Batch - July 2024.
 // Copyright (c) Metamation India.
+// Kalaimagal V P
 // ------------------------------------------------------------------------------------------------
 // Test.c
-// Program on main branch.
+// Program on A5 branch.
+// Write a program that sorts integer arrays in ascending order using any of the sorting algorithms discussed
+// in the Sorting workshop. Also, implement binary search algorithm to search for an element in the sorted array.
+// You may take the arrays as input from the user.
 // ------------------------------------------------------------------------------------------------
 #include <stdio.h>
-#include "Header.h"
+#include <ctype.h>    // For tolower
+#include <stdlib.h>   // For strtol and realloc
+#include <conio.h>    // For getch()
+#include "header.h"
 
+// ANSI escape codes for colors
+#define RESET   "\033[0m"
+#define SKYBLUE "\033[36m"
+#define GREEN   "\033[32m"
+#define YELLOW  "\033[33m"
+#define RED     "\033[31m"
+
+// Function to clear the console screen
+void ClearScreen () {
+#ifdef _WIN32
+   system ("cls");           // Windows-specific command to clear the screen
+#endif
+}
+
+// Function to format and print test results
+void PrintTestCase (int numTests, int inputs[], int actualResults) {
+   int sortedArray[50];
+   // Copy input array for sorting
+   for (int i = 0; i < actualResults; i++) sortedArray[i] = inputs[i];
+   InsertionSort (sortedArray, actualResults);
+   printf (YELLOW "------------------ Test Case %d -------------------\n" RESET, numTests);
+   printf ("| Input Array:            | ");
+   for (int i = 0; i < actualResults; i++) printf ("%d  ", inputs[i]);
+   printf ("|\n| Output Array:           | ");
+   for (int i = 0; i < actualResults; i++) printf ("%d  ", sortedArray[i]);
+   printf ("|\n  Status: %s\n", GREEN "PASS THE SORTED ARRAY" RESET);
+   printf ("--------------------------------------------------\n\n");
+}
+
+// Test sorting
+void TestSort () {
+   struct {

```

Comment:
Here expected refers to just the length of array, how it can be used to define whether an array is sorted.